### PR TITLE
change to the `.format()` function

### PR DESCRIPTION
**Pull request by MarkusHackspacher**
Change to the .format() function in some file in the  horizons/world/ direction.

### CODE REVIEW

File: horizons/world/units/collectors/collector.py [hunk 1]
```diff
@@ -396,8 +396,8 @@ def transfer_res_from_target(self):
 				new_reslist.append( entry )
 
 			remnant = self.get_component(StorageComponent).inventory.alter(entry.res, actual_amount)
-			assert remnant == 0, "%s couldn't take all of res %s; remnant: %s; planned: %s" % \
-			       (self, entry.res, remnant, entry.amount)
+			assert remnant == 0, "{} couldn't take all of res {}; remnant: {}; planned: {amount}".format(
```
Comment (squiddy) [inline horizons/world/units/collectors/collector.py]: Why did you name make the last argument (`amount`) to format? Just use `{}` like the others.

File: horizons/world/units/collectors/collector.py [hunk 2]
```diff
@@ -406,8 +406,8 @@ def transfer_res_to_home(self, res, amount):
 		remnant = self.get_home_inventory().alter(res, amount)
 		#assert remnant == 0, "Home building could not take all resources from collector."
 		remnant = self.get_component(StorageComponent).inventory.alter(res, -amount)
-		assert remnant == 0, "%s couldn't give all of res %s; remnant: %s; inventory: %s" % \
-		       (self, res, remnant, self.get_component(StorageComponent).inventory)
+		assert remnant == 0, "{} couldn't give all of res {}; remnant: {}; inventory: {inv}".format(
```
Comment (squiddy) [inline horizons/world/units/collectors/collector.py]: Why did you name make the last argument (`inv`) to format? Just use `{}` like the others.

**Comment by AndyMender** [comment]
What about using `{:s}` instead of `{}`? It's not necessary, because Python defaults to string literals when the substitution field is left empty? Similar cases for floats and decimals were raised [here](https://github.com/unknown-horizons/unknown-horizons/pull/2617).

**Comment by squiddy** [comment]
> What about using {:s} instead of {}?

I think we've decided somewhere in an earlier PR that we just use `{}`.

> Similar cases for floats and decimals were raised here.

In these cases, the previous code used explicit `%d` format modifiers, and initially these were changed to `{}`, which changed the behaviour.

**Comment by AndyMender** [comment]
I'll bear this in mind, thank you for the clarification! :)

**Pull request merged**